src/command_modules/azure-cli-redis/azure/cli/command_modules/redis/custom.py

```diff
@@ -108,20 +78,64 @@ def cli_redis_create(cmd, client,
     :param enable_non_ssl_port: If the value is true, then the non-ssl redis server port (6379) will be enabled.
     :param tenant_settings: Json dictionary with tenant settings
     :param shard_count: The number of shards to be created on a Premium Cluster Cache.
+    :param minimum_tls_version: Optional: requires clients to use a specified
+     TLS version (or higher) to connect (e,g, '1.0', '1.1', '1.2'). Possible
+     values include: '1.0', '1.1', '1.2'
     :param subnet_id: The full resource ID of a subnet in a virtual network to deploy the redis cache in. Example format /subscriptions/{subid}/resourceGroups/{resourceGroupName}/Microsoft.{Network|ClassicNetwork}/VirtualNetworks/vnet1/subnets/subnet1
     :param static_ip: Required when deploying a redis cache inside an existing Azure Virtual Network.
+    :param zones: A list of availability zones denoting where the resource needs to come from.
+    :param tags: Json dictionary with Resource tags. Example : {\"testKey\":\"testValue\"}
     """
     from azure.mgmt.redis.models import RedisCreateParameters, Sku
+    # pylint: disable=too-many-function-args
     params = RedisCreateParameters(
-        location=location,
-        sku=Sku(sku, vm_size[0], vm_size[1:]),
-        tags=tags,
+        Sku(sku, vm_size[0], vm_size[1:]),
```

Comment:
Do you think some validation that the vm_size applies to the SKU would be helpful? If the service end validation / error is clear, then this is unnecessary, but could still save the CLI from sending a request doomed to fail.